src/uploads/uploads.service.ts

```diff
@@ -7,6 +7,8 @@ import { TestCase } from './testCase.entity';
 import { Upload } from './upload.entity';
 
 export interface UploadsFilter {
+  owner: string;
```

Comment:
I think it's best they are kept separate for url encoding. The entire slug can't be url encoded because it will also encode the slash.

Reply:
We've been handling it [like this](https://github.com/ably-labs/test-observability-server/blob/6718ce7a4fcef9650fc11ab5a8fb776d58cb02bf/src/utils/viewModel/urlHelpers.ts#L12-L14) elsewhere, would that not be enough?

Reply:
Depends how correct you want to be.

The slug `A%2fB/C` will be decoded into `A/B/C` and then spit and encoded as `A/B/C`.

My thinking was typescript aims to do things more correct so I should try and code that way too. Also not thinking about stuff like this can often lead to vulns.

Reply:
Sorry, I'm a bit confused by the situation you're describing above. Are you describing a situation where you have a repo whose owner's name is "A%2fB", and repository name is "C"?

If so, then the `repoName` parameter passed to the linked `encodeRepoName` function would be `A%2fB/C`, and the function would return `A%252fB/C`, which I think is what we'd want.

Reply:
No because the webserver will unescapee the %2 back into a / before passing it to your code.

Reply:
In URL encoding, a character is always represented by two hexadecimal digits. `%2` is not a valid URL-encoded character. I believe the server will, in fact, unescape the `%25` into a `%`.

Reply:
`%2f`* I missed the f.

Reply:
But, thinking about it, I don't think I'm actually hugely fussed by the thing I originally commented about, so I'm gonna resolve this one anyway.

Reply:
But, thinking about it, I don't think I'm actually hugely fussed by the thing I originally commented about, so I'm gonna resolve this one anyway.